display_books_promise.js

```diff
@@ -0,0 +1,66 @@
+let BookData = [];
+
+const insertDataInHtml = (data) => {
+  data.sort((a, b) => a.price - b.price);
+  const tableBody = document.getElementById('table-all-books');
+  tableBody.innerHTML = '';
+  const headerRow = document.createElement('tr');
+  headerRow.innerHTML = `
+    <th>BookId</th>
+    <th>Genre</th>
+    <th>Price</th>
+    <th>Examine</th>
+   `;
+  tableBody.appendChild(headerRow);
+  data.forEach((book) => {
+    const row = document.createElement('tr');
+    row.innerHTML = `
+       <th>${book.bookId}</th>
+       <th>${book.genre}</th>
+       <th>${book.price}</th>
+      `;
+    tableBody.appendChild(row);
+  });
+};
+
+fetch('random_books.json')
+  .then((response) => response.json())
+  .then((data) => {
+    BookData = data;
+    insertDataInHtml(BookData);
+  })
+  .catch((error) => console.log(error));
+
+const searchByIdButton = (event) => {
+  event.preventDefault();
+  const idValue = document.getElementById('input-search-id').value;
+  if (!idValue.trim().length) return;
+  const matchingBooks = BookData.filter(
+    (book) => book.bookId.toString() === idValue,
+  );
+  insertDataInHtml(matchingBooks);
+};
+
+const searchByGenreButton = (event) => {
```

Comment:
Instead of event use evt. 
event is not a reserved keyword, but it is a global variable in IE at least.